frontend/src/app/(pages)/profile/page.tsx

```diff
@@ -1,9 +1,69 @@
+"use client"
 import ProfileComponent from "@/components/profile/Profile"
+import { UserService } from "@/helpers/user/user_api_wrappers";
+import User from "@/types/user";
+import { Metadata } from "next"
+import { FormEvent, useEffect, useState } from "react";
+import { toast } from "react-toastify";
 
-export default function ProfilePage() {
+export const metadata: Metadata = {
+    title: "Profile",
+    description: "profile information",
+}
+
+export default async function ProfilePage() {
+
+    const [user, setUser] = useState<User | undefined>(undefined);
+
+    const email = typeof window !== "undefined" ? sessionStorage.getItem("email") || "" : "";
+
+    useEffect(() => {
+        fetchUser();
+    }, [])
+
+    const fetchUser = async () => {
+        const rawUser = await UserService.getUserByEmail(email);
+        console.log("User retrieved: " + JSON.stringify(rawUser));
+        setUser(rawUser);
+    }
+    
+    async function saveInformation(e: FormEvent<HTMLFormElement>, updatedUser: User, preferences: Preference) {
```

Comment:
I have a question here. Why don't we just define this `saveInformation` function in the `Information` component instead of passing it through multiple component layers? Just curious as I might oversee something.

Reply:
Ok true that. I think i was too focused on trying to make the async components in the page.tsx that i didnt move it back. Done